### PR TITLE
fix ups converted RM13 combat armor

### PR DESCRIPTION
**Pull request by Terrorforge**
#### Summary
Bugfixes "fixes RM13 Combat Armor not working with UPS conversion"

#### Purpose of change
RM13 Combat Armor refuses to turn on when converted to UPS. Spawn a UPS (w/ battery), RM13 and UPS Conversion Mod, install the mod. It will allow you try to turn it on, but will say the fuel cells are dead.

#### Describe the solution

Added a check to the function so it only exits and gives the "dead cells" message if the armor has no charge AND it's not a UPS-converted armor with enough UPS charge to run.

#### Describe alternatives you've considered

There probably needs to be a larger-scale reworking of powered items so this doesn't need to be done individually for every single one, i.e. #48608, but this is a much quicker fix that will get this particular item working right away.

#### Testing

1. Spawn a UPS (w/ battery), RM13 and UPS Conversion Mod
2. Install UPS Conversion Mod in RM13
3. Turn on RM13

### CODE REVIEW

File: src/iuse.cpp [hunk 1]
```diff
@@ -2062,7 +2062,7 @@ cata::optional<int> iuse::extinguisher( player *p, item *it, bool, const tripoin
 cata::optional<int> iuse::rm13armor_off( player *p, item *it, bool, const tripoint & )
 {
     // This allows it to turn on for a turn, because ammo_sufficient assumes non-tool non-weapons need zero ammo, for some reason.
-    if( !it->ammo_sufficient() ) {
+    if( !it->ammo_sufficient() && !( it->has_flag( flag_USE_UPS ) && p->has_enough_charges( *it, false )) ) {
```
Comment (Saicchi) [inline src/iuse.cpp:2065]:
```suggestion
    if( !it->ammo_sufficient() && !( it->has_flag( flag_USE_UPS ) &&
                                     p->has_enough_charges( *it, false ) ) ) {
```

**Review by Saicchi — CHANGES_REQUESTED** [review]

**Comment by andrei8l** [comment]
Using the UPS conversion mod with the RM13 is technically a bug, probably introduced by this other [bugfix](https://github.com/CleverRaven/Cataclysm-DDA/pull/41599).

**Comment by Terrorforge** [comment]
> Using the UPS conversion mod with the RM13 is technically a bug, probably introduced by this other [bugfix](https://github.com/CleverRaven/Cataclysm-DDA/pull/41599).

Hrm. Well, this fix is probably still preferable to the current situation where installing the mod permanently destroys the armor. You could just disable it entirely, but I have no idea how to do that.

I'd argue it's also preferable to the "intended" behavior, because in my experience the batteries it's supposed to use and the recipe for crafting them are essentially non-existent.

UPS conversion is probably a little too convenient, but I'd say it's an acceptable stopgap until the other problems can be addressed.

**Comment by andrei8l** [comment]
> You could just disable the modification entirely, but I have no idea how to do that.

Easiest way is to add an `IIREMOVABLE` dummy toolmod, then modify the RM13 spawn group to include that toolmod (by adding `contents-item` to its entry). This works but it's quite ugly and will break in the future if tools are allowed to have multiple mods.

Another hacky way is to add an extra flag check in the filtering lambda in `iuse::toolmod_attach()`, then add the new flag to the RM13's JSON definition.

The best way is only a little more involved. You'd have to first modify `Item_factory::load( islot_mod &slot, ...)` to load a new blacklist array (probably still with a flag rather than item ids), then integrate that into `iuse::toolmod_attach`.

**Comment by Hirmuolio** [comment]
Does the `"valid_mod_locations"` that is used on guns work on other items?
Some new mod locations could be defined for all the electronic mods and then the slots would have to be defined on all electronic items.

**Comment by I-am-Erk** [comment]
Any intention to pick this one back up?

**Comment by Terrorforge** [comment]
> Any intention to pick this one back up?

It's been fixed by the rework I mentioned in the OP. I converted an RM13 to UPS just the other day.

Hot tip, though: unless you have the internal UPS CBM and want to run it off bionic power, just convert it to using heavy batteries directly. The RM13 doesn't have any pockets, and the UPS is too bulky to fit in anything but a backpack.

**Comment by Night-Pryanik** [comment]
@Terrorforge could you please resolve conflicts?

**Comment by Terrorforge** [comment]
> @Terrorforge could you please resolve conflicts?

No, this fix is no longer relevant. I'm closing this.

**Pull request closed without merging**